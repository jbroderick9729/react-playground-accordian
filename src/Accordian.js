import React, { Component } from 'react';

export default class Accordian extends Component {
    static defaultProps = {
        sections: []
    }

    state = {
        tabIndex: null
    }

    handleClick(index) {        
        this.setState({
            tabIndex: index
        })
    }
    
    renderButtons = () => {
        return this.props.sections.map((section, index) => 
            <li key={index}>
                <button onClick={() => this.handleClick(index)}>
                    {section.title}
                </button>
                <p>
                    { (this.state.tabIndex === index) && section.content}
                </p>
            </li>)
    }

    render() {        
        return (
            <div>
                <ul>
                    {this.renderButtons()}
                </ul>
            </div>
        )
    }
}





