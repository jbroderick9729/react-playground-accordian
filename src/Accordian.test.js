import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian from './Accordian'

const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

describe('Accordian component', () => {
    
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordian />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders empty li when sections prop not supplied', () => {
        const wrapper = shallow(<Accordian />);
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it('renders no sections as active by default', () => {
        const wrapper = shallow(<Accordian sections={sections} />);
        expect(toJson(wrapper)).toMatchSnapshot()
    });

})