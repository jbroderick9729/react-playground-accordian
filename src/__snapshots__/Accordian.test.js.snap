// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian component component opens a clicked section 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
      <p />
    </li>
  </ul>
</div>
`;

exports[`Accordian component only opens the last section when multiple sections have been clicked 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <p />
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
      <p>
        Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
      </p>
    </li>
  </ul>
</div>
`;

exports[`Accordian component renders empty li when sections prop not supplied 1`] = `
<div>
  <ul />
</div>
`;

exports[`Accordian component renders no sections as active by default 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <p />
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
      <p />
    </li>
  </ul>
</div>
`;
